### PR TITLE
For review: Initial Influxdb 0.9.x metrics impl

### PR DESCRIPTION
**Pull request by msavy**
I haven't finished writing tests yet, but want to put this up for review to see what @EricWittmann and @KurtStam think.

### CODE REVIEW

**Comment by msavy** [comment]
BTW, I did fixup on a bunch of commits, so the commit dates are inaccurate :o)

File: gateway/engine/core/src/main/java/io/apiman/gateway/engine/metrics/impl/influxdb/InfluxDb09Metrics.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                    throw (InfluxException) result.getError();
+                }
+                endSignal.countDown();
+            }
+        });
+
+        try {
+            endSignal.await();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (EricWittmann) [inline gateway/engine/core/src/main/java/io/apiman/gateway/engine/metrics/impl/influxdb/InfluxDb09Metrics.java]: Boy do we need need a comprehensive logging strategy!

**Comment by EricWittmann** [comment]
This look pretty good (and very straightforward).  There are probably opportunities to optimize (as we've already discussed) but it's a good start.  I'm just waiting for your go-ahead before I merge this PR.

**Comment by KurtStam** [comment]
Do you have a picture of the high level architecture? Also some javadoc may come in handy :).

**Comment by msavy** [comment]
Ok, feel free to merge, and I've noted and will make those improvements suggested later. Note that given influx is now on RC13 things might change at some point soon.

**Pull request merged**